### ISSUE TITLE
Incorrect error when derived class appears first in an ambient namespace

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#10045]
```ts
declare namespace ns {
  class SecondNS extends FirstNS { }
  class FirstNS { }
}
```
Actual: Error
Expected: No error
